rosbridge_library/src/rosbridge_library/internal/ros_loader.py

```diff
@@ -177,26 +179,46 @@ def _load_class(modname, subname, classname):
     more expressive exceptions.
 
     Returns the loaded module, or None on failure """
-    global loaded_modules
 
     try:
         with _manifest_lock:
             # roslib maintains a cache of loaded manifests, so no need to duplicate
-            roslib.launcher.load_manifest(modname)
+            _load_manifest(modname)
     except Exception as exc:
         raise InvalidPackageException(modname, exc)
 
     try:
-        pypkg = __import__('%s.%s' % (modname, subname))
+        pypkg = importlib.import_module('%s.%s' % (modname, subname))
     except Exception as exc:
         raise InvalidModuleException(modname, subname, exc)
 
     try:
-        return getattr(getattr(pypkg, subname), classname)
+        return getattr(pypkg, classname)
     except Exception as exc:
         raise InvalidClassException(modname, subname, classname, exc)
 
 
+def _load_manifest(modname):
+    """
+    Simplified version of roslib's load_manifest.
+
+    Updates system path to include the path of a given package.
+    """
+    global _loaded_modules
+
+    if modname in _loaded_modules:
+        return
+
+    prefix_path = get_prefix_path(modname)
+    package_path = os.path.join(prefix_path, 'lib', 'python3.{}'.format(sys.version_info.minor), 'site-packages')
```

Comment:
This logic is not portable. What is this needed for? All Python packages should already be in the path.

Reply:
The original version of `rosbridge_library` makes use of `roslib` to load packages that might not be in the path, so I wanted to have a minimal version of that functionality so that the `ros2` version works as the `ros1` version as much as possible.

If it's safe to assume that all Python packages are in the path, then this code can be removed and we can just import the required packages directly. I'll do some tests just to double check.

Reply:
I ended up removing this part; as you indicated above it should be safe to assume that the packages are in the path already.